nitro-protocol/contracts/examples/HashLockedSwap.sol

```diff
@@ -2,61 +2,76 @@
 pragma solidity 0.7.6;
 pragma experimental ABIEncoderV2;
 
-import '../interfaces/IForceMoveApp.sol';
 import {ExitFormat as Outcome} from '@statechannels/exit-format/contracts/ExitFormat.sol';
+import {StrictTurnTaking} from '../libraries/signature-logic/StrictTurnTaking.sol';
+import '../interfaces/IForceMoveApp.sol';
 
 /**
- * @dev The HashLockedSwap contract complies with the ForceMoveApp interface and implements a HashLockedSwaped payment
+ * @dev The HashLockedSwap contract complies with the ForceMoveApp interfaces and strict turn taking logic and implements a HashLockedSwaped payment.
  */
 contract HashLockedSwap is IForceMoveApp {
     struct AppData {
         bytes32 h;
         bytes preImage;
     }
 
+    /**
+     * @notice Decodes the appData.
+     * @dev Decodes the appData.
+     * @param appDataBytes The abi.encode of a AppData struct describing the application-specific data.
+     * @return AppData struct containing the application-specific data.
+     */
+    function appData(bytes memory appDataBytes) internal pure returns (AppData memory) {
+        bytes memory decodedAppData = abi.decode(appDataBytes, (bytes));
+        return abi.decode(decodedAppData, (AppData));
```

Comment:
Yes, it is. This 'decoding of decoded bytes' has appeared when we at Openware first used the nitro-protocol and had a persisting 'Invalid signer' error. It was decided to get around this issue by introducing an additional step of encoding, so that State hash in contracts would be the same as one in go-nitro.
It was meant to be a temporary workaround, but nothing lasts longer than the temporary 😄

Reply:
I have removed all additional encodings from [`state.getVariablePart`](https://github.com/statechannels/go-nitro/blob/main/nitro-protocol/src/contract/state.ts#L61),[`state.encodeState`](https://github.com/statechannels/go-nitro/blob/main/nitro-protocol/src/contract/state.ts#L100) and [`embeddedApplication.getVariablePart`](https://github.com/statechannels/go-nitro/blob/main/nitro-protocol/src/contract/embedded-application.ts#L17).
Thus, additional step of decoding was also removed from `HashLockSwap` and `CountingApp`.